horace_core/sqw/@sqw/set_efix.m

```diff
@@ -23,53 +23,79 @@
 
 % Perform operations
 % ==================
-narg=numel(varargin);
-if narg<1 || narg>2
+if ~(isnumeric(efix) && numel(efix)>=1 && all(isfinite(efix)) && all(efix>=0))
     error('HORACE:sqw:invalid_argument',...
-        'This function accepts one or two input arguments')
+        'efix must be numeric scalar or array of finite values');
 end
-if narg>=1
-    efix=varargin{1}(:);
-    if ~(isnumeric(efix) && numel(efix)>=1 && all(isfinite(efix)) && all(efix>=0))
-        error('HORACE:sqw:invalid_argument',...
-            'efix must be numeric scalar or array of finite values');
-    end
-end
-if narg>=2
-    emode=varargin{2};
+if ~exist('emode','var')
+    emode=[];   % indicates emode to be left untouched
+else
     if ~(isnumeric(emode) && isscalar(emode) && (emode==0||emode==1||emode==2))
         error('HORACE:sqw:invalid_argument',...
             'emode must 1 (direct geometry), 2 (indirect geometry) or 0 (elastic)')
```

Comment:
```suggestion
            'emode must be one of: 0 (elastic), 1 (direct geometry), or 2 (indirect geometry) ')
```

Reply:
more grammatically correct, but in reality we use 1 almost all of the time. Seen usage of 2 once... But changed.